bokehjs/src/coffee/renderer/glyph/glyph.coffee

```diff
@@ -71,11 +72,35 @@ class GlyphView extends ContinuumView
 
     @_map_data()
 
+  project_xy: (x, y) ->
+    merc_x_s = []
+    merc_y_s = []
+    for i in [0...x.length]
+      [merc_x, merc_y] = proj4(toProjection, [x[i], y[i]])
+      merc_x_s[i] = merc_x
+      merc_y_s[i] = merc_y
+    return [merc_x_s, merc_y_s]
+
+  project_xsys: (xs, ys) ->
+    merc_xs_s = []
+    merc_ys_s = []
+    for i in [0...xs.length]
+      [merc_x_s, merc_y_s] = @project_xy(xs[i], ys[i])
+      merc_xs_s[i] = merc_x_s
+      merc_ys_s[i] = merc_y_s
+    return [merc_xs_s, merc_ys_s]
+
   set_data: (source) ->
     # set all the coordinate fields
     for name, prop of @coords
       @[name] = prop.array(source)
 
+    if @renderer.plot_model.use_map
+      if @x?
+        [@x, @y] = @project_xy(@x, @y)
+      if @xs?
+        [@xs, @ys] = @project_xsys(@xs, @ys)
+
```

Comment:
Definitely! And probably put it in a set of geo utils. Also see related ideas in #2861.